chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -508,38 +578,82 @@ func (b *BitcoindNotifier) RegisterSpendNtfn(outpoint *wire.OutPoint,
 	// is registered, the output hasn't already been spent. If the output
 	// is no longer in the UTXO set, the chain will be rescanned from the point
 	// where the output was added. The rescan will dispatch the notification.
-	txout, err := b.chainConn.GetTxOut(&outpoint.Hash, outpoint.Index, true)
+	txOut, err := b.chainConn.GetTxOut(&outpoint.Hash, outpoint.Index, true)
 	if err != nil {
 		return nil, err
 	}
 
-	if txout == nil {
-		// TODO: fall back to scanning blocks if txindex isn't on.
-		transaction, err := b.chainConn.GetRawTransactionVerbose(&outpoint.Hash)
+	if txOut == nil {
+		tx, err := b.chainConn.GetRawTransactionVerbose(&outpoint.Hash)
 		if err != nil {
+			// Avoid returning an error if the transaction was not
+			// found.
 			jsonErr, ok := err.(*btcjson.RPCError)
 			if !ok || jsonErr.Code != btcjson.ErrRPCNoTxInfo {
+				return nil, fmt.Errorf("unable to query for "+
+					"txid %v: %v", outpoint.Hash, err)
+			}
+		}
+
+		var blockHash *chainhash.Hash
+		if tx != nil && tx.BlockHash != "" {
+			// If we're able to retrieve a valid block hash from the
+			// transaction, then we'll use it as our rescan starting
+			// point.
+			blockHash, err = chainhash.NewHashFromStr(tx.BlockHash)
+			if err != nil {
 				return nil, err
 			}
+		} else {
+			// Otherwise, we'll rely on manually scanning our chain
```

Comment:
Why's this block necessary? Instead we can just query for the block hash of the height hint. Then the block below can be simplified to simply call into `Rescan` with the proper block hash.

Reply:
Good point, will fix.

Reply:
Fixed.